galley/pkg/fs/fsutilities.go

```diff
@@ -0,0 +1,79 @@
+//  Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fs
+
+import (
+	"bytes"
+	"crypto/sha1"
+	"reflect"
+
+	"github.com/ghodss/yaml"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type istioResource struct {
+	u   *unstructured.Unstructured
+	sha [sha1.Size]byte
+}
+
+type fileResourceKey struct {
+	key  string
+	kind string
+}
+
+func (r *istioResource) key() string {
+	if len(r.u.GetNamespace()) > 0 {
+		return r.u.GetNamespace() + "/" + r.u.GetName()
```

Comment:
This is prone to creating garbage. Consider calculating it once and storing it.

Reply:
done, move to istioResource